src/backend/distributed/commands/distribute_object_ops.c

```diff
@@ -1769,6 +1791,16 @@ GetDistributeObjectOps(Node *node)
 					return &TextSearchDict_Comment;
 				}
 
+				case OBJECT_DATABASE:
+				{
+					return &Database_Comment;
+				}
+
+				case OBJECT_ROLE:
```

Comment:
The deparse and qualify functions for comments on DATABASE and ROLE are pretty much exactly the same, and they are also the same for TextSearchConfig_Comment and TextSearchDict_Comment too. Let's stop copy pasting them and create a reusable function. Eventually we'll want to support comments for all types of objects. So let's prepare the code a bit so that that won't involve having to copy paste functions around.

Reply:
pinging @thanodnl and @hanefi in case they disagree, because they wrote the COMMENT implementations for `TextSearchConfig` and `TextSearchDict`

Reply:
I made it as reusable as possible I don't think that it can be more reusable other than this structure

Reply:
When we check Rename structure as well, it has different implementations for each statement type

Reply:
I don't think you understood what I meant with my initial comment. So I'll try to be clearer this time. The DatabaseCommentObjectAddress and RoleCommentObjectAddress are **exactly** the same, except for an Assert. If you remove the Assert you can reuse a single function for both.

And while the deparse functions (`DeparseDatabaseCommentStmt`, and `DeparseRoleCommentStmt`) don't have completely identical bodies, most of it is the same. The only thing that is different is the stringified name of the object type. So doing something like this should work fine:
```c
const char * const ObjectTypeNames[] =
{
    [OBJECT_DATABASE] = "DATABASE",
    [OBJECT_ROLE] = "ROLE",
    [OBJECT_TSCONFIGURATION]  = "TEXT SEARCH CONFIGURATION",
    [OBJECT_TSDICTIONARY]  = "TEXT SEARCH DICTIONARY",
    /* etc. */
};

char *
DeparseCommentStmt(Node *node)
{
	CommentStmt *stmt = castNode(CommentStmt, node);
	StringInfoData str = { 0 };
	initStringInfo(&str);

	char const *objectType = ObjectTypeNames[stmt->objtype];
	char const *objectName = quote_identifier(strVal(stmt->object));

	char *comment = stmt->comment != NULL ? quote_literal_cstr(stmt->comment) : "NULL";

	appendStringInfo(&str, "COMMENT ON %s %s IS %s;", objectType, objectName, comment);

	return str.data;
}
```

And once you are using the same function for deparse and qualify, you can notice that the definitions of the `Database_Comment` and `Role_Comment` are identical too. So then you can create a single shared one, e.g. called `Any_Comment`.

> When we check Rename structure as well, it has different implementations for each statement type

Possibly that code could be simplified in a similar manner too then. But even if that's not the case, in general, having lots of copy pasted code in one place is not really an argument to do the same in a different place.

Reply:
Sidenote regarding simplifying RENAME: This seems harder to do than what I'm suggesting for COMMENT simplification. The main reason is that with RENAME allows renaming "nested" objects, such as columns in a table or constraints inside a domain.

Reply:
I got your point. Let me check it

Reply:
This is a great idea Jelte. I liked your suggestion as well.

I suggest we add an assertion that `ObjectTypeNames[stmt->objtype]` is not null so that we see assertions when we want to add support for comments on a new database objects if we forger to add the name in the array.

Reply:
@JelteF Fixed with some modifications. Could you check?